components/raftstore/src/store/fsm/peer.rs

```diff
@@ -5897,8 +5915,14 @@ where
         }
         self.fsm.peer.size_diff_hint = 0;
```

Comment:
If we do not reset it here, seems the split check would simply retry to avoid the split propose is failed for some reason?

Reply:
Do you mean do not reset `compaction_declined_bytes` or `size_diff_hint`? 

Yes, if we keep them unreset, split check will re-run.  But we do need to reset them here otherwise if the split check says it's not splitable, there's no point to run the split-check again. And also these two variables do not cover the SST ingestion scenario as they might be all zero.  So we still need may_skip_split_check. 

Today we use three variables to trigger split check:
1) size_diff_hint: if the write/delete is big enough, trigger split check.
2) compaction_declined_bytes: after compaction is done, the enough diff size triggers split check to update the approximate size. 
3) may_skip_split_check:  trigger the split check after SST ingestion is done. 

I cannot think of anything that can simplify the code without impacting the correctness and efficiency.

Reply:
To prevent the unnecessary split check task run is critical, as it's very expensive.